src/editors/containers/VideoEditor/components/VideoSettingsModal/components/ThumbnailWidget/hooks.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { actions } from '../../../../../../data/redux';
+
+export const fileInput = ({ setThumbnailSrc }) => {
+  const dispatch = useDispatch();
+  const ref = React.useRef();
+  const click = () => ref.current.click();
+  const addFile = (e) => {
+    const file = e.target.files[0];
+    const reader = new FileReader();
+    const image = file;
+    reader.onload = () => {
+      setThumbnailSrc(reader.result);
+    };
+    if (image) {
+      reader.readAsDataURL(image);
+      dispatch(actions.video.updateField({ thumbnail: image }));
```

Comment:
are we not also dispatching to a backend thumbnail endpoint?

Reply:
The ticket says that the thumbnail should only be saved to the block when all the settings are saved. So I only need to update the redux store.

Reply:
`saved to the block`  is there also an external api that handles the file?

Reply:
https://2u-internal.atlassian.net/wiki/spaces/TNL/pages/62291971/TNL-9760+Document+api+endpoints+and+data+contracts+for+all+video+settings+events#Add-a-thumbnail might need to be addressed